docs/config.md

```diff
@@ -0,0 +1,64 @@
+# DefraDB configuration (YAML)
+
+The default DefraDB directory is `$HOME/.defradb`. It can be changed via the --rootdir CLI flag.
+
+Relative paths are interpreted as being rooted in the DefraDB directory.
+
+## `datastore.store`
+
+Store can be badger or memory. Defaults to `badger`.
+
+- badger: fast pure Go key-value store optimized for SSDs (https://github.com/dgraph-io/badger)
+- memory: in-memory version of badger
```

Comment:
thought: `memory` here confuses me a little bit:

1. We have our own memory store, and I though that was what `datastore.store=memory` was
2. Badger memory is driven by the badger memory boolean option, if I am incorrect with (1), then why are we duplicating the option

suggestion: Unless I'm really confused/incorrect here, and given that we do document some badger config options in this file, I think it might be worth documenting the badger memory config prop.

Reply:
Ah fair enough.

> We could swap it to the memory datastore, but I'd prefer to do that in a new PR.

I agree that *if* we want to make that change it should be done in another PR.